src/Sulu/Bundle/CategoryBundle/Controller/CategoryController.php

```diff
@@ -230,93 +190,138 @@ public function putAction($id, Request $request)
-            $view = $this->view($e->toArray(), 404);
-        } catch (KeyNotUniqueException $e) {
-            $view = $this->view($e->toArray(), 400);
+        $listBuilder = $this->initializeListBuilder($locale);
+
+        if (!$request->get('search')) {
+            $listBuilder->where($this->getManager()->getFieldDescriptor($locale, 'parent'), $parentId);
         }
 
-        return $this->handleView($view);
+        // need to disable default limit because some frontend components are not paginated
+        if ($disableLimit || !$request->get('limit')) {
+            $listBuilder->limit(null);
+        }
+
+        $results = $listBuilder->execute();
+
+        if ($expandIds) {
+            // expand the paths to the categories which are assigned to the given expandIds.
+            // we need to do this in a second query, because we need to expand children of the current page only.
+            $resultIds = array_map(
+                function ($category) {
+                    return $category['id'];
+                },
+                $results
+            );
+            $results = array_merge($results, $this->expandChildren($expandIds, $resultIds, $locale));
+        }
+
+        foreach ($results as &$result) {
+            $result['hasChildren'] = ($result['lft'] + 1) !== $result['rgt'];
+        }
+
+        $list = new CategoryListRepresentation(
+            $results,
+            self::$entityKey,
+            'get_categories',
+            $request->query->all(),
+            $listBuilder->getCurrentPage(),
+            $listBuilder->getLimit(),
+            $listBuilder->count()
+        );
+
+        return $list;
     }
 
     /**
-     * Returns a Category-list-representation.
+     * Initializes and returns a ListBuilder instance which is used when returning a CategoryListRepresentation
+     * for the given locale. The returned ListBuilder is initialized with the request-parameters and respective
+     * select fields.
      *
-     * @param Request $request
-     *
-     * @throws MissingParameterException iff the locale is missing in the request
+     * @param $locale
      *
-     * @return CategoryListRepresentation
+     * @return DoctrineListBuilder
      */
-    protected function getCategoryListRepresentation(Request $request, $parentKey = null)
+    private function initializeListBuilder($locale)
     {
         /** @var RestHelperInterface $restHelper */
-        $restHelper = $this->getRestHelper();
+        $restHelper = $this->get('sulu_core.doctrine_rest_helper');
 
         /** @var DoctrineListBuilderFactory $factory */
         $factory = $this->get('sulu_core.doctrine_list_builder_factory');
 
-        $listBuilder = $factory->create(self::$entityName);
-        $locale = $this->getRequestParameter($request, 'locale', true);
         $fieldDescriptors = $this->getManager()->getFieldDescriptors($locale);
-        $listBuilder->sort($fieldDescriptors['depth']);
```

Comment:
yes, you are right. i just saw that this frontend bug was fixed with https://github.com/massiveart/husky/pull/451. so it should not be neccesary to sort by depth anymore.